domains/discount-domain.js

```diff
@@ -41,6 +41,20 @@ function validateCoupon (coupon, price, item, now = new Date()) {
       error: 'Coupon expired'
     }
   }
+  if (coupon.conditions) {
+    for (let i = 0; i < coupon.conditions.length; i++) {
```

Comment:
As logic in this is now complicated, we can notice it from multiple indent. It is a smell to refactor to other functions and unit tested separately